### ISSUE TITLE
[Adjust Rule] SIM909

### ISSUE THREAD

**Issue by sondrelg** [MartinThoma/flake8-simplify#129]
## Desired change

* **Rule(s)**: SIM909
* **Adjustment**: Consider allowing declarations on classes

## Explanation

I have this code:

```python
database = Database(url=url)
metadata = sqlalchemy.MetaData()

class BaseMeta:
    metadata = metadata
    database = database
```

The `BaseMeta` class is used for database models as the `Meta` baseclass.

This is currently flagged by SIM909, but after reading through the rationale of the rule, I think this could be considered a false positive. Assignment of `metadata` to `metadata` on a class is not the same as outside a class right - it actually serves a purpose 🙂

What do you think?

**Comment by MartinThoma** [comment]
You're absolutely right. SIM909 needs a fix. I hope in 1-2 hours I can release a new version with the fix.

Thank you for reporting it!

**Comment by MartinThoma** [comment]
## Python

```python
class BaseMeta:
    metadata = metadata
```

## AST

```
$ python -m astpretty --no-show-offsets /dev/stdin <<< `cat example.py`
Module(
    body=[
        ClassDef(
            name='BaseMeta',
            bases=[],
            keywords=[],
            body=[
                Assign(
                    targets=[Name(id='metadata', ctx=Store())],
                    value=Name(id='metadata', ctx=Load()),
                    type_comment=None,
                ),
            ],
            decorator_list=[],
        ),
    ],
    type_ignores=[],
)
```

**Comment by sondrelg** [comment]
Let me know when the new release is out and I can test on my repo if you'd like 🙂

**Comment by MartinThoma** [comment]
[`flake8-simplify==0.19.2`](https://pypi.org/project/flake8-simplify/#history) was just released with the fix :-)

**Comment by sondrelg** [comment]
All the (now) false positives are gone, and it seems to run well. Thanks @MartinThoma!

**Comment by MartinThoma** [comment]
Nice! Thank you for reporting the issue and checking if the fix works!
